crates/conf/src/macros.rs

```diff
@@ -0,0 +1,142 @@
+        use $crate::io::load_conf_text;
+        use tracing::error;
+        use paste::paste;
+        use thiserror::Error as ErrorDerive;
+        use bevy::prelude::Resource;
+
+        #[derive(Debug, ErrorDerive)]
+        pub enum ConfigLoadError {
+            #[error("Failed to load configuration: {0}")]
+            Error(Error),
+            #[error("Configuration check failed: {0:?}")]
+            CheckErrors(Vec<(String, Error)>),
+        }
+
+        impl From<Error> for ConfigLoadError {
+            fn from(err: Error) -> Self {
+                Self::Error(err)
+            }
+        }
+
+        #[derive(Resource, Debug, Clone)]
+        pub struct Configuration {
+            $(
+                $name: $type_into,
+            )*
+        }
+
+        #[derive(Debug, Clone)]
+        pub struct FromConfiguration {
+            $(
+                $name: $type_from,
+            )*
+        }
+
+        impl TryInto<Configuration> for FromConfiguration {
+            type Error = Error;
+
+            fn try_into(self) -> Result<Configuration, Self::Error> {
+                Ok(Configuration {
+                    $(
+                        $name: self.$name.try_into()?,
+                    )*
+                })
+            }
+        }
+
+        #[derive(Deserialize, Debug, Clone, Default)]
+        struct PartialConfiguration {
+            $(
+                $name: Option<paste! {[<Partial $type_from>]}>,
+            )*
+        }
+
+        impl Configuration {
+            $(
+                pub fn $name(&self) -> &$type_into {
+                    &self.$name
+                }
+            )*
+
+            pub async fn load(path: &Path) ->  Result<Self, ConfigLoadError> {
+                let from = FromConfiguration::load(path).await?;
+                Ok(from.try_into().unwrap())
+            }
+        }
+
+        impl FromConfiguration {
+            pub async fn load(path: &Path) ->  Result<Self, ConfigLoadError> {
+                match load_conf_text(path).await? {
+                    Some(text) => {
+                        let partial: PartialConfiguration =
+                            serde_yaml::from_str(text.as_str()).context("Failed to parse DE configuration")?;
+                        let config: Self = partial.try_into().context("Failed to convert partial configuration")?;
+
+                        let mut errors = vec![];
+                        $(
+                            let value = &config.$name;
+                            let check = value.check();
+                            if let Err(err) = check {
+                                for err in err {
+                                    error!("Failed check: {:?}", err);
+                                    errors.push(err);
+                                }
+                            }
+                        )*
+                        if !errors.is_empty() {
+                            error!("Failed checks:");
```

Comment:
Ditto, I would change this to trace or remove it whatsoever. `ConfigLoadError::CheckErrors` should be comprehensively handled in `plugin.rs` after the loading task returns.